src/bootstrap/bin/rustc.rs

```diff
@@ -175,6 +175,9 @@ fn main() {
         if let Ok(s) = env::var("RUSTC_CODEGEN_UNITS") {
             cmd.arg("-C").arg(format!("codegen-units={}", s));
         }
+        if stage != "0" && env::var("RUSTC_THINLTO").is_ok() {
+            cmd.arg("-Ccodegen-units=16").arg("-Zthinlto");
```

Comment:
Could we respect the `rust.codegen-units` settings in `config.toml` here instead of hard-coding "16"?

Reply:
This is all intended to be pretty temporary, and for now I'd prefer to just keep this easy to remove later as I don't think anyone's really configuring codegen units anyway.

Reply:
@alexcrichton  Ok, please add a FIXME.

(I did set the CGU to 3 😆)